gensim/models/wrappers/ldamallet.py

```diff
@@ -245,11 +248,40 @@ def show_topic(self, topicid, topn=10):
     def print_topic(self, topicid, topn=10):
         return ' + '.join(['%.3f*%s' % v for v in self.show_topic(topicid, topn)])
 
+
+    def get_version(self, direc_path):
+        """"
+
+        function to return the version of `mallet`
+
+        """
+        try:
+            """
+            Check version of mallet via jar file
+            """
+            archive = zipfile.ZipFile(direc_path, 'r')
+            if u'cc/mallet/regression/' not in archive.namelist():     
+                return '2.0.7'
+            else:
+                return '2.0.8RC3'
+        except Exception:
+            
+            xml_path = direc_path.split("bin")[0]
+            try:
+                doc = et.parse(xml_path + "pom.xml").getroot()
+                namespace = doc.tag[:doc.tag.index('}') + 1]
+                return doc.find(namespace + 'version').text.split("-")[0]
+            except Exception:
+                return "No version file to detect"
+        
+
+
     def read_doctopics(self, fname, eps=1e-6, renorm=True):
         """
         Yield document topic vectors from MALLET's "doc-topics" format, as sparse gensim vectors.
 
         """
+        mallet_version = self.get_version(self.mallet_path)
```

Comment:
If you can, then yes. 
But both versions can give dense output not sure how it will help with `if len(parts) == 2*self.num_topics:`